### PR TITLE
Add a parser performance test

### LINKED ISSUE #4053: Parser performance tests
### Problem

We need a way to test the performance of the parser at speeds greater than what consensus nodes support.

### Solution

* Add properties to generate a mix of transaction types at different speeds
* Calculate number of transactions to generate that TPS to fill a two second record file
* Generate the transactions and add to record file
* Send to `RecordFileParser.parse()`

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by steven-sheehy**
**Description**:
* Add a `RecordFileBuilder` for programmatic generation of large record files
* Add `hedera.mirror.importer.parser.record.performance` for declarative generation of performance test scenarios
* Add a `RecordFileParserPerformanceTest` that can test ingest performance of a remote database using the above

**Related issue(s)**:

Fixes #4053

**Notes for reviewer**:

**Checklist**

- [x] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: hedera-mirror-importer/src/test/java/com/hedera/mirror/importer/parser/domain/RecordFileBuilder.java [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+                case CRYPTOAPPROVEALLOWANCE:
+                    return recordItemBuilder::cryptoApproveAllowance;
+                case CRYPTOCREATEACCOUNT:
+                    return recordItemBuilder::cryptoCreate;
+                case CRYPTODELETE:
+                    return recordItemBuilder::cryptoDelete;
+                case CRYPTODELETEALLOWANCE:
+                    return recordItemBuilder::cryptoDeleteAllowance;
+                case CRYPTOTRANSFER:
+                    return recordItemBuilder::cryptoTransfer;
```
Comment (steven-sheehy) [inline hedera-mirror-importer/src/test/java/com/hedera/mirror/importer/parser/domain/RecordFileBuilder.java:191]: I could not find an easy way to construct this mapping at runtime without resorting to slower reflection-based approaches. So we'll have to remember to update this mapping going forward whenever we add a new `RecordItemBuilder` builder method.

File: hedera-mirror-importer/src/test/java/com/hedera/mirror/importer/parser/domain/RecordItemBuilder.java [hunk 2]
```diff
@@ -354,6 +376,10 @@ public Builder<NodeStakeUpdateTransactionBody.Builder> nodeStakeUpdate() {
         return new Builder<>(TransactionType.NODESTAKEUPDATE, builder);
     }
 
+    public Builder<PrngTransactionBody.Builder> prng() {
```
Comment (steven-sheehy) [inline hedera-mirror-importer/src/test/java/com/hedera/mirror/importer/parser/domain/RecordItemBuilder.java:379]: Going forward we should always have a no-arg builder for any arg-based builder

**Review by edwin-greene — APPROVED** [review]
LGTM!

File: hedera-mirror-importer/src/test/java/com/hedera/mirror/importer/parser/domain/RecordFileBuilder.java (hunk 1 above)
Comment (steven-sheehy) [reply hedera-mirror-importer/src/test/java/com/hedera/mirror/importer/parser/domain/RecordFileBuilder.java]: Nevermind, found a way.

**Review by edwin-greene — APPROVED** [review]
LGTM!

**Review by xin-hedera — APPROVED** [review]
looking good!

**Pull request merged**